packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,278 @@
+    ThisIsSilent:
+        ${GetParameters} $PassedInParameters
+        ${GetOptions} $PassedInParameters "/RemoteURL" $RemoteUrl
+        ${GetOptions} $PassedInParameters "/RemoteUser" $RemoteUser
+        ${GetOptions} $PassedInParameters "/RemotePassword" $RemotePassword
+        ${GetOptions} $PassedInParameters "/EnableExporter" $EnableExporterValue
+        Call Install
+        # TODO add error conditions
+        Return
+    RunInstaller:
+        Return
+SectionEnd
+
+
+Section "install"
+    IfSilent ThisIsSilent RunInstaller
+    ThisIsSilent:
+        Return
+    RunInstaller:
+        Call Install
+SectionEnd
+
+Function remoteWriteInfo
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+    ${NSD_CreateLabel} 0 0 100% 12u "Enter URL for remote write"
+    Pop $RemoteWriteLabel
+
+    ${NSD_CreateText} 0 13u 100% 12u ""
+    Pop $RemoteWriteText
+
+    ${NSD_CreateLabel} 0 26u 100% 12u "Enter User for remote write"
+    Pop $RemoteWriteUserLabel
+
+    ${NSD_CreateText} 0 39u 100% 12u ""
+    Pop $RemoteWriteUserText
+
+    ${NSD_CreateLabel} 0 52u 100% 12u "Enter Password for remote write"
+    Pop $RemoteWritePasswordLabel
+
+    ${NSD_CreateText} 0 65u 100% 12u ""
+    Pop $RemoteWritePasswordText
+
+    nsDialogs::Show
+FunctionEnd
+
+
+Function remoteWriteInfoLeave
+    ${NSD_GetText}  $RemoteWriteText  $RemoteUrl
+    ${NSD_GetText}  $RemoteWriteUserText  $RemoteUser
+    ${NSD_GetText}  $RemoteWritePasswordText  $RemotePassword
+FunctionEnd
+
+Function enableWindowsExporter
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+
+    ${NSD_CreateLabel} 0 0 100% 12u "Enable Windows Exporter"
+    Pop $EnableExporterLabel
+
+    ${NSD_CreateCheckBox} 0 13u 100% 12u ""
+    Pop $EnableExporterCheck
+
+    nsDialogs::Show
+FunctionEnd
+
+Function enableWindowsExporterLeave
+    ${NSD_GetState} $EnableExporterCheck $EnableExporterValue
+    ${If} $EnableExporterValue == ${BST_CHECKED}
+        StrCpy $EnableExporterValue "true"
+    ${Else}
+        StrCpy $EnableExporterValue "false"
+    ${EndIf}
+FunctionEnd
+
+
+Function .onInit
+    setShellVarContext all
+    !insertmacro VerifyUserIsAdmin
+FunctionEnd
+
+# Uninstaller
+Function un.onInit
+    SetShellVarContext all
+
+    #Verify the uninstaller - last chance to back out
+    MessageBox MB_OKCANCEL "Permanantly remove ${APPNAME}?" IDOK next
+        Abort
+    next:
+    !insertmacro VerifyUserIsAdmin
+FunctionEnd
+
+Function Install
+    # Premptively stop Grafana Agent if it is running
+    nsExec::ExecToLog 'sc stop "Grafana Agent"'
+    Pop $0
+    # Files for the install directory - to build the installer, these should be in the same directory as the install script (this file)
+    setOutPath $INSTDIR
+    # Files added here should be removed by the uninstaller (see section "uninstall")
+    file "agent-windows-amd64.exe"
+    file "logo.ico"
```

Comment:
It won't let me comment on the logo icon itself since it's a binary file :) 

It doesn't seem to be a square image, so it'll get stretched out and mess up the aspect ratio. Can you add whitespace to make it square?

Reply:
Sorry, I was making too many assumptions here 🙃 In the image you shared from the setup, the aspect ratio of the icon seemed off: 

![](https://raw.githubusercontent.com/grafana/agent/1ead68d4670ea777480a3536e5987adf17751b46/docs/assets/remote_options.png)

Compared to the robot in the logo:

![](https://raw.githubusercontent.com/grafana/agent/1ead68d4670ea777480a3536e5987adf17751b46/docs/assets/logo_and_name.png)

Reply:
Good eye, the converter squashed it around, but I redid it and its now the sqautter? icon.

Reply:
![Screen Shot 2021-03-31 at 10 34 27 AM](https://user-images.githubusercontent.com/3036838/113161819-bb9fb180-920c-11eb-889a-b94411eff399.png)

Reply:
Perfect! Thanks!